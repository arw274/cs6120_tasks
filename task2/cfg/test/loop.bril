# 0 -> {1}, 1 -> {3, 4}, 2 -> {3}, 3 -> {1}, 4 -> {}
@main (input: int) {
    one: int = const 1;
    cond0: bool = ge input one;
.loop:
    br cond0 .ge .zero;
    b: int = const one;
.ge:
    input: int = sub input one;
    jmp .loop;
.zero:
    ret;
}