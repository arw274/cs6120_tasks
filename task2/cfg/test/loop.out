Block 0: [{'dest': 'one', 'op': 'const', 'type': 'int', 'value': 1}, {'args': ['input', 'one'], 'dest': 'cond0', 'op': 'ge', 'type': 'bool'}]
Block 1: [{'label': 'loop'}, {'args': ['cond0'], 'labels': ['ge', 'zero'], 'op': 'br'}]
Block 2: [{'args': ['one'], 'dest': 'b', 'op': 'const', 'type': 'int'}]
Block 3: [{'label': 'ge'}, {'args': ['input', 'one'], 'dest': 'input', 'op': 'sub', 'type': 'int'}, {'labels': ['loop'], 'op': 'jmp'}]
Block 4: [{'label': 'zero'}, {'op': 'ret'}]
{'main': {0: [1], 1: [3, 4], 2: [3], 3: [1], 4: []}}
